fix TURN/STURN

xundeenergie opened the pull request:
* remove docker-compose-coturn.yml and replace it with and example-file, because there are user-data inside, which should not be on git
* add a whole rtc_config_example-coturn.json to be copied to rtc_config.json and only the domain <DOMAIN> should be changed.
* modified the documentation
* modified the .gitignore to ignore the files with user-data

schlagmichdoch commented on turnserver_example.conf:
```diff
@@ -35,4 +38,5 @@ cert=/etc/letsencrypt/live/<DOMAIN>/cert.pem
 
 # 443 for TURN over TLS, which can bypass firewalls
+# the standard-port is 5349
 tls-listening-port=443
```
shouldn't that better be `5349` to be consistent with `rtc_config_example-coturn.json`. It could then say `Port for TURN over TLS`. Usage of 443 can bypass firewalls.

schlagmichdoch commented on docs/host-your-own.md:
```diff
@@ -397,9 +397,26 @@ Now point your browser to `http://localhost:8080`.
+- To stop the container including coturn run `docker-compose -f docker-compose-coturn.yml stop`
+
+## Firewall
+To run PairDrop including its own coturn-server you need to punch holes in the firewall. This ports must be opened additionally:
```
`These` ports must be opened additionally:

schlagmichdoch commented on docs/host-your-own.md:
```diff
@@ -397,9 +397,26 @@ Now point your browser to `http://localhost:8080`.
+- copy `rtc_config_example-coturn.json` to `rtc_config.json`
+- copy `turnserver_example.conf` to `turnserver.conf`
+- change <DOMAIN> in all three files to the domain, where your pairdrop is running
+- change user and password for turn-server in `turnserver.conf` and `rtc-config.json`
```
Probably, there should be another step that you need to generate ssl certificates

xundeenergie replied:
I changed this to the new improvements and added this step.

schlagmichdoch commented on docker-compose-coturn.yml:
```diff
@@ -10,10 +10,18 @@ services:
     network_mode: "host"
     volumes:
       - ./turnserver.conf:/etc/coturn/turnserver.conf
+      - ./logs/:/var/log/
```
as your logging to STDOUT now, you don't need that anymore, do you?

xundeenergie replied:
this should be removed now.

schlagmichdoch commented on .gitignore:
```diff
@@ -3,3 +3,9 @@ node_modules
+logs/*
+*.orig
+*.log
+letsencrypt/*
```
We don't need lines 8-10. Lines 3-5 could also be deleted but maybe in a separate commit

xundeenergie replied:
I have them removed now.

schlagmichdoch reviewed (changes requested):
Thanks a lot for this PR! I had just some small change requests. Also, some your commits appear to be authored with your real name. If you don't want it to appear in the contributors list you might rebase your commit before merging.

xundeenergie commented:
I've just seen... with working TLS for turns and stuns i can see also over the internet every user, which is on pairdrop.schuerz.at

Is this, how it should work? It thought, only if devices are paired, you can see each other...

xundeenergie commented:
Now i reverted all this trials to server coturn also on port 443 and changed it to port 5349 for tls-secured connections.
Now it should work as easy as possible to run coturn for pairdrop configured and ssl-secured with turns and stuns almost out of the box.
